helper/run-in-nix-env

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
```

Comment:
How about we add a header that documents what this script does?

Reply:
+1. Done.

Reply:
`run-in-nix-env` was needlessly convoluted, I've cleaned it up.